### PR TITLE
[JENKINS-73669] don't change unrelated checkboxes in rowSelectionCont…

### PR DESCRIPTION
**Pull request by mawinter69**
when a row in table contains more than 1 checkbox the rowSelectionController shouldn't change all checkboxes but only the ones that match a given class.

See [JENKINS-73669](https://issues.jenkins.io/browse/JENKINS-73669).

### Testing done
Tested in context of https://github.com/jenkinsci/agent-maintenance-plugin/pull/213

### Proposed changelog entries

- don't change unrelated checkboxes in rowSelectionController

### Proposed upgrade guidelines

N/A

```[tasklist]
### Submitter checklist
- [ ] The Jira issue, if it exists, is well-described.
- [ ] The changelog entries and upgrade guidelines are appropriate for the audience affected by the change (users or developers, depending on the change) and are in the imperative mood (see [examples](https://github.com/jenkins-infra/jenkins.io/blob/master/content/_data/changelogs/weekly.yml)). Fill in the **Proposed upgrade guidelines** section only if there are breaking changes or changes that may require extra steps from users during upgrade.
- [ ] There is automated testing or an explanation as to why this change has no tests.
- [ ] New public classes, fields, and methods are annotated with `@Restricted` or have `@since TODO` Javadocs, as appropriate.
- [ ] New deprecations are annotated with `@Deprecated(since = "TODO")` or `@Deprecated(forRemoval = true, since = "TODO")`, if applicable.
- [ ] New or substantially changed JavaScript is not defined inline and does not call `eval` to ease future introduction of Content Security Policy (CSP) directives (see [documentation](https://www.jenkins.io/doc/developer/security/csp/)).
- [ ] For dependency updates, there are links to external changelogs and, if possible, full differentials.
- [ ] For new APIs and extension points, there is a link to at least one consumer.
```

### Desired reviewers

@mention

Before the changes are marked as `ready-for-merge`:

```[tasklist]
### Maintainer checklist
- [ ] There are at least two (2) approvals for the pull request and no outstanding requests for change.
- [ ] Conversations in the pull request are over, or it is explicit that a reviewer is not blocking the change.
- [ ] Changelog entries in the pull request title and/or **Proposed changelog entries** are accurate, human-readable, and in the imperative mood.
- [ ] Proper changelog labels are set so that the changelog can be generated automatically.
- [ ] If the change needs additional upgrade steps from users, the `upgrade-guide-needed` label is set and there is a **Proposed upgrade guidelines** section in the pull request title (see [example](https://github.com/jenkinsci/jenkins/pull/4387)).
- [ ] If it would make sense to backport the change to LTS, a Jira issue must exist, be a _Bug_ or _Improvement_, and be labeled as `lts-candidate` to be considered (see [query](https://issues.jenkins.io/issues/?filter=12146)).
```

### CODE REVIEW

**Review by NotMyFault — APPROVED** [review]

File: core/src/main/resources/hudson/PluginManager/updates.jelly [hunk 1]
```diff
@@ -69,7 +69,7 @@ THE SOFTWARE.
                 <tr>
                   <l:isAdmin>
                     <th data-sort-disable="true">
-                      <l:rowSelectionController>
+                      <l:rowSelectionController class="checkbox-install-plugin">
```
Comment (timja) [inline core/src/main/resources/hudson/PluginManager/updates.jelly:72]: should be namespaced as an internal class with the app- prefix I think
```suggestion
                      <l:rowSelectionController class="app-checkbox-install-plugin">
```

**Review by timja — APPROVED** [review]
I assume this isn’t used elsewhere? Another option would be to by default do all but allow narrowing it?

Approved anyway

**Comment by mawinter69** [comment]
> I assume this isn’t used elsewhere? Another option would be to by default do all but allow narrowing it?

I intend to use it, see link above

**Review by SNanda8895 — APPROVED** [review]
There is no security problem here.

**Pull request merged**